packages/sdk/src/Fuel.ts

```diff
@@ -1,16 +1,40 @@
+import { WalletLocked } from 'fuels';
 import type { AbstractAddress } from 'fuels';
 
 import { FuelWalletConnection } from './FuelWalletConnection';
-import { FuelWalletLocked } from './FuelWalletLocked';
 import { FuelWalletProvider } from './FuelWalletProvider';
 
+// Isolate the provider instance to prevent
+// developers from replacing the provider
+// instance with a new one
+const FuelWeb3Privates: {
+  provider?: FuelWalletProvider;
+} = {};
+
 export class Fuel extends FuelWalletConnection {
-  getWallet(address: string | AbstractAddress): FuelWalletLocked {
-    return new FuelWalletLocked(address, this);
+  async getProvider(): Promise<FuelWalletProvider> {
+    const providerConfig = await this.network();
+    // Return the current provider instance if it exists
+    if (FuelWeb3Privates.provider) {
+      return FuelWeb3Privates.provider;
+    }
+    // Otherwise, create a new provider instance
+    // fetch the current network and connect the provider
+    const provider = new FuelWalletProvider(providerConfig.url, this);
+    FuelWeb3Privates.provider = provider;
+
+    // Listen for network changes and connect the provider
+    // selected network from the user
+    this.on('network', async (network) => {
```

Comment:
Would it be possible for the network to change without a call to `getProvider`?  Basically I'm asking would it be possible for a network change event to fire before reaching this code which handles network change events.  Sorry if this is a dumb question lol.

Reply:
As the getProvider always returns the same instance when the event is triggered and the URL has changed, the changes will have an instant effect.

If the event is triggered before riching this point would not be a problem as the current network is requested from the wallet at the time we create the instance of the provider line 16.